url78eec0e0b6_Orikuro_vic_tgz-pJ8-net_nexon_vindictus_itemcomparer_inport_StaticImportJ8/src/net/nexon/vindictus/itemcomparer/logic/ThreadStarter.java

```diff
@@ -0,0 +1,311 @@
+package net.nexon.vindictus.itemcomparer.logic;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.OutputStreamWriter;
+import java.text.DecimalFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+import net.nexon.vindictus.itemcomparer.modell.Combo;
+import net.nexon.vindictus.itemcomparer.modell.ItemSet;
+import net.nexon.vindictus.itemcomparer.modell.Items;
+import net.nexon.vindictus.itemcomparer.modell.ext.Armor;
+import net.nexon.vindictus.itemcomparer.modell.ext.Gloves;
+import net.nexon.vindictus.itemcomparer.modell.ext.Helm;
+import net.nexon.vindictus.itemcomparer.modell.ext.Pants;
+import net.nexon.vindictus.itemcomparer.modell.ext.Shoes;
+
+/*
+ * 
+ */
+
+public class ThreadStarter {
+	private String PARAMETER = "";
+
+	public ThreadStarter(String[] args, int cpus, int results, double price,
+			int atk, int matk, Comparator<Combo> defc, Items items,
+			boolean noinfo, boolean nocsv) {
+		PARAMETER = Arrays.toString(args).replace(",", "").replace("[", "")
+				.replace("]", "");
+
+		start(cpus, results, price, atk, matk, defc, items, noinfo, nocsv);
+	}
+
+	private void start(int cpus, int results, double price, int atk, int matk,
+			Comparator<Combo> defc, Items items, boolean noinfo, boolean nocsv) {
+		start(cpus, results, price, atk, matk, defc, items.getShoes(),
+				items.getPants(), items.getGloves(), items.getArmors(),
+				items.getHelms(), noinfo, nocsv);
+	}
+
+	private StringBuffer sb = new StringBuffer();
+	private DecimalFormat df = new DecimalFormat(",###");
+
+	private void start(int cpus, int results, double price, int atk, int matk,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 16. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 68-207 into a separate method.